keel-core/src/test/kotlin/com/netflix/spinnaker/keel/events/ResourceEventSerializationTests.kt

```diff
@@ -0,0 +1,119 @@
+package com.netflix.spinnaker.keel.events
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceMetadata
+import com.netflix.spinnaker.keel.api.ResourceName
+import com.netflix.spinnaker.keel.api.SPINNAKER_API_V1
+import com.netflix.spinnaker.keel.api.randomUID
+import com.netflix.spinnaker.keel.persistence.randomData
+import com.netflix.spinnaker.keel.serialization.configuredObjectMapper
+import com.netflix.spinnaker.keel.serialization.configuredYamlMapper
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+import strikt.jackson.has
+import strikt.jackson.path
+import strikt.jackson.textValue
+import java.time.Clock
+import java.time.Instant
+import java.time.ZoneId
+
+internal class ResourceEventSerializationTests : JUnit5Minutests {
```

Comment:
Polymorphism +  Jackson deserialization is always a barrel of laughs so I put some tests around it.